src/snowflake/snowpark/dataframe.py

```diff
@@ -1600,7 +1600,7 @@ def join(
             right: The other :class:`Dataframe` to join.
             using_columns: A list of names of the columns, or the column objects, to
                 use for the join.
-            join_type: The type of join ("inner", "full", "left", "right").
+            join_type: The type of join ("inner", "full", "left", "right", "semi", "anti", "cross").
```

Comment:
all allowed join type strs are here https://github.com/snowflakedb/snowpark-python/blob/2fb6672b9c5851f6098a01e55e64b1d7ffc2fa84/src/snowflake/snowpark/_internal/analyzer/binary_plan_node.py#L10-L24

Reply:
Yeah I thought about it but I'm not sure if we want to add duplicate like `full` and `fullouter`, so I just followed the existing pattern where each category only contains the shorter one. But if this is preferred I could do a full copy for sure.

Reply:
probably we can mentioned them all (like fullouter and rightouter are both supported by snowflake https://docs.snowflake.com/en/sql-reference/constructs/join.html#syntax), and I believe some like leftsemi and semi are added to be compatible with pyspark. But it's also worthwhile mentioning that some strs are just aliases of another one, e.g., `leftanti` is an alias of `anti` (you can find all of them here https://github.com/snowflakedb/snowpark-python/blob/2fb6672b9c5851f6098a01e55e64b1d7ffc2fa84/src/snowflake/snowpark/_internal/analyzer/binary_plan_node.py#L30-L48)

Reply:
Yeah let me make it more organized, such as:

```
We support the following join types:

Left outer join: "left" or "leftouter"
...
```

What do you think?

Reply:
yes, looks perfect